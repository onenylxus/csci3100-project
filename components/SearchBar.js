// Import
import React from 'react';
import { TextInput, View } from 'react-native';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import Style from '../assets/style';

// Export Search Bar
export default function SearchBar() {
  // const [searchWord, setsearchWord] = React.useState('');

  return (
    <View style={Style.SectionStyle}>
      <FontAwesomeIcon
        icon={faSearch}
        size={15}
        style={{ marginHorizontal: 5, marginVertical: 12 }}
      />
      <TextInput
        placeholder="Search"
        style={{ outline: 'none', width: '100%' }}
        // onChangeText={(text) => setsearchWord(text)}
      />
    </View>
  );
}
