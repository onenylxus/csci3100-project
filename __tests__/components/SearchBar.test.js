// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import SearchBar from '../../components/SearchBar';

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Run jest tests
describe('SearchBar', () => {
  let element;

  beforeEach(() => {
    element = render(<SearchBar />);
  });

  it('compiles successfully', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
