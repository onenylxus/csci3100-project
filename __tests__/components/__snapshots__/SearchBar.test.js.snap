// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar compiles successfully 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#fff",
      "borderColor": "#000",
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flexDirection": "row",
      "height": 40,
      "marginTop": 10,
      "width": "95%",
    }
  }
>
  <
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f002",
          "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
        ],
        "iconName": "search",
        "prefix": "fas",
      }
    }
    size={15}
    style={
      Object {
        "marginHorizontal": 5,
        "marginVertical": 12,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search"
    rejectResponderTermination={true}
    style={
      Object {
        "width": "85%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f057",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
          ],
          "iconName": "times-circle",
          "prefix": "fas",
        }
      }
      size={15}
      style={
        Object {
          "marginHorizontal": 5,
          "marginVertical": 12,
        }
      }
    />
  </View>
</View>
`;
